tests/test_noninteractive_mode.py

```diff
@@ -0,0 +1,49 @@
+""" Non-interactive tests. """
+import subprocess
+from mssqltestutils import create_mssql_cli
+
+def test_session_closure_query_valid():
+    """ Test session closure for valid query. """
+    assert is_processed_closed("select 1")
+
+def test_session_closure_query_invalid():
+    """ Test session closure for invalid query. """
+    assert is_processed_closed("asdfasoifjas")
+
+def test_query_large():
```

Comment:
>test_query_large [](start = 4, length = 16)

I feel like you have two sets of tests: ones that use MssqlCli and ones that execute the process. Can we break these up and put them into classes the leverage the test setup and teardown methods to assert pre/post condition? For example, a common post condition in the MssqlCli client test is to assert the sqltoolsservice process is closed, which could be in the teardown.

Reply:
Good idea!